### PR TITLE
[datadog_logs_index] Add daily limit reset options to log indexes

### PR DESCRIPTION
**Pull request by Prepager**
Description
---
This PR adds support for daily limit reset options as previously added in: https://github.com/DataDog/datadog-api-spec/pull/2622

- `daily_limit_reset`
  - `reset_time`
  - `reset_utc_offset`
- `daily_limit_warning_threshold_percentage`

Public API documentation: https://docs.datadoghq.com/api/latest/logs-indexes/

### CODE REVIEW

File: datadog/resource_datadog_logs_index.go [hunk 1]
```diff
@@ -28,18 +29,43 @@ var indexSchema = map[string]*schema.Schema{
...
+		Optional:         true,
+		MaxItems:         1,
+		DiffSuppressFunc: suppressDiffWhenDisabledDailyLimit,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"reset_time": {
+					Description:  "String in `HH:00` format representing the time of day the daily limit should be reset. The hours must be between 00 and 23 (inclusive).",
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([0-1][0-9]|2[0-3]):00$`), "must be in HH:00 format with the hours (HH) between 00 and 23 (inclusive)"),
```
Comment (Prepager) [inline datadog/resource_datadog_logs_index.go:50]: Regex validation test cases playground: https://regex101.com/r/s6CIVF/1

File: datadog/resource_datadog_logs_index.go [hunk 2]
```diff
@@ -28,18 +29,43 @@ var indexSchema = map[string]*schema.Schema{
...
+				},
+				"reset_utc_offset": {
+					Description:  "String in `(-|+)HH:00` format representing the UTC offset to apply to the given reset time. The hours must be between -12 and +14 (inclusive).",
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringMatch(regexp.MustCompile(`^(-(0[0-9]|1[0-2])|\+(0[0-9]|1[0-4])):00$`), "must be in +HH:00 or -HH:00 format with the hours (HH) between -12 and +14 (inclusive)"),
```
Comment (Prepager) [inline datadog/resource_datadog_logs_index.go:56]: Regex validation test cases playground: https://regex101.com/r/TNPPpl/1

**Review by artslidd — APPROVED** [review]
Changes look good to me ✅

**Review by HantingZhang2 — APPROVED** [review]

**Review by aliciascott — APPROVED** [review]
looks good

**Review by capucinemartin — APPROVED** [review]

**Comment by Zhann** [comment]
Is it possible this sets a permadiff on resources with `daily_limit` set, but not `daily_limit_reset`?

See https://github.com/DataDog/terraform-provider-datadog/issues/2278

**Pull request merged**